tests/ImageSharp.Tests/MemoryAllocatorValidator.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Diagnostics;
+using SixLabors.ImageSharp.Diagnostics;
+using Xunit;
+
+namespace SixLabors.ImageSharp.Tests
+{
+    public static class MemoryAllocatorValidator
+    {
+        public static IDisposable MonitorAllocations(int max = 0)
+        {
+            MemoryDiagnostics.Current = new();
+            return new TestMemoryAllocatorDisposable(max);
+        }
+
+        public static void ValidateAllocation(int max = 0)
```

Comment:
Is there a use case for `max > 0`? Do we want to allow any test with monitoring enabled to leak *some* memory?